packages/tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,297 @@
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import Tab, { TabProps } from './Tab';
+import TabTitle from './TabTitle';
+import omit from 'lodash/omit';
+
+const borderHeight = 3;
+
+const listStyle = css`
+  list-style: none;
+  padding: 0px;
+  display: flex;
+  width: 100%;
+`;
+
+const activeStyle = css`
+  color: ${uiColors.gray.dark3};
+  cursor: default;
+`;
+
+const disabledStyle = css`
+  pointer-events: none;
+  color: ${uiColors.gray.light2};
+`;
+
+const grayBorder = css`
+  height: ${borderHeight}px;
+  background-color: ${uiColors.gray.light2};
+  position: relative;
+  // Makes border overlap with box-model for TabTitle components
+  margin-top: -${borderHeight}px;
+  pointer-events: none;
+`;
+
+const greenBorder = css`
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  background-color: ${uiColors.green.base};
+  transition: 150ms transform ease-in-out, 150ms width ease-in-out 10ms;
+`;
+
+const focusedStyle = css`
+  background-color: ${uiColors.blue.base};
+`;
+
+function isTab(
+  element: React.ReactNode,
+): element is React.ReactElement<TabProps, typeof Tab> {
+  if (!(element != null && typeof element === 'object' && 'type' in element)) {
+    return false;
+  }
+
+  return (element.type as any).displayName === 'Tab';
+}
+
+interface TabsProps {
+  /**
+   * Content that will appear inside of Tabs component. Should be comprised of at least two Tabs.
+   */
+  children: Array<React.ReactElement>;
+
+  /**
+   * Callback to be executed when Tab is selected. Receives index of activated Tab as the first argument.
+   */
+  setSelected?: any;
+
+  /**
+   * Index of the Tab that should appear active. If value passed to selected prop, component will be controlled by consumer.
+   */
+  selected?: number;
+
+  /**
+   * className supplied to Tabs container.
+   */
+  className?: string;
+
+  /**
+   * HTML Element that wraps title in Tab List.
+   */
+  as?: React.ElementType<any>;
+}
+
+/**
+ * # Tabs
+ *
+ * Tabs component
+ *
+ * ```
+<Tabs selected={0} setSelected={() => execute callback when new Tab is selected}>
+  <Tab name='First Tab'>Tab 1</Tab>
+  <Tab name='Second Tab'>Tab 2</Tab>
+</Tabs>
+```
+ * @param props.children Content to appear inside of Tabs component.
+ * @param props.setSelected Callback to be executed when Tab is selected. Receives index of activated Tab as the first argument.
+ * @param props.selected Index of the Tab that should appear active. If value passed, component will be controlled by consumer.
+ * @param props.className className applied to Tabs container.
+ * @param props.as HTML Element that wraps name in Tab List.
+ */
+function Tabs({
+  children,
+  setSelected,
+  selected,
+  className,
+  as = 'button',
+  ...rest
+}: TabsProps) {
+  const isControlled = typeof selected === 'number';
+  const childrenArray = React.Children.toArray(children);
+  const [activeIndex, setActiveIndex] = useState(
```

Comment:
Seems like a relic to call this activeIndex - should be selectedIndex.

Outside of that, this is to hold the controlled value right? I'm wondering if we can handle all that behavior forking up at the top of the function. Perhaps alias the selected prop to `controlledSelected` and rename this to `uncontrolledSelected`.

Then decide which one is the source of truth at the beginning of the function and assign that to `selected` so all code can reference that one value.

Same thing for the update function.

If you want to pair on this, I'm happy to as well. I hope it makes the code cleaner, but wouldn't be surprised if it doesn't as well.

Reply:
👍